suse/factory.go

```diff
@@ -0,0 +1,238 @@
+package suse
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strings"
+	"sync"
+
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/quay/claircore/pkg/ovalutil"
+	"github.com/quay/zlog"
+	"golang.org/x/net/html"
+)
+
+//doc:url updater
+const base = `https://ftp.suse.com/pub/projects/security/oval/`
+
+var (
+	reELFile   = regexp.MustCompile(`suse.linux.enterprise.server.([1-9][1-9]).xml.gz`)
+	reLeapFile = regexp.MustCompile(`opensuse.leap.(1[0-9].[0-9]+).xml.gz`)
+)
+
+type Factory struct {
+	c    *http.Client
+	base *url.URL
+
+	mu   sync.Mutex
+	etag string
+	cur  driver.UpdaterSet
+}
+
+// UpdaterSet implements [driver.UpdaterSetFactory].
+func (f *Factory) UpdaterSet(ctx context.Context) (driver.UpdaterSet, error) {
+	ctx = zlog.ContextWithValues(ctx, "component", "suse/Factory.UpdaterSet")
+	us := driver.NewUpdaterSet()
+	if f.c == nil {
+		zlog.Info(ctx).
+			Msg("unconfigured")
+		return us, nil
+	}
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, f.base.String(), nil)
+	if err != nil {
+		return us, fmt.Errorf("suse: unable to construct request: %w", err)
+	}
+	f.mu.Lock()
+	defer f.mu.Unlock()
+	if f.etag != "" {
+		req.Header.Set(`if-none-match`, f.etag)
+	}
```

Comment:
I don't think this is ever set? Also, I didn't see an etag returned when I curl'd an example endpoint

Reply:
Fair enough, looks like the initial updaterset creation cannot rely on not-modifed directives, luckily the OVAL files themselves include etags.

Reply:
updated